src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinNamesAnnotationIntrospector.kt

```diff
@@ -111,11 +112,13 @@ internal class KotlinNamesAnnotationIntrospector(
         }
     }
 
-    override fun hasCreatorAnnotation(member: Annotated): Boolean =
-        if (member is AnnotatedConstructor && member.isKotlinConstructorWithParameters())
-            cache.checkConstructorIsCreatorAnnotated(member) { hasCreatorAnnotation(it) }
-        else
-            false
+    override fun findCreatorAnnotation(config: MapperConfig<*>, ann: Annotated): JsonCreator.Mode? {
+        if (ann !is AnnotatedConstructor || !ann.isKotlinConstructorWithParameters()) return null
```

Comment:
So Kotlin does not add factory methods as creators, just constructors?

Reply:
Yes, the role of this function is to consider the primary constructor as a `JsonCreator` when there is no function with `JsonCreator` assigned.
Factory methods are not processed here because they are explicitly granted `JsonCreator`.

Reply:
Ah. Going forward (2.17?) I hope to add a new type of Creator level -- "preferred" (or "default" or something -- which is higher than implicit (auto-detected) but lower than explicit annotation.
The idea is that it would be used as THE Creator if no annotations used, but actual `@JsonCreator` annotation could be used to override. But the two would not conflict -- unlike the case of 2 annotated constructors, for example, there would be clear precedence (explicit annotation wins). Similarly "regular" auto-detection could find other candidates (from Java side), but modules (Kotlin, Scala) could indicate something that is automatically preferred.
This would also work better for Java Records which have the "canonical constructor" which should be indicated as "preferred" over alternatives.